lib/open_mic.rb

```diff
@@ -0,0 +1,26 @@
+require './lib/joke.rb'
+require './lib/user.rb'
+
+class OpenMic
+  attr_reader :location, :date, :performers, :told_jokes
+
+  def initialize(location_and_date_hash)
+    @location = location_and_date_hash[:location]
+    @date = location_and_date_hash[:date]
+    @performers = []
+  end
+
+  def welcome(performer)
+    @performers << performer
+  end
+
+  def repeated_jokes?
+    open_mic_jokes = []
+    @performers.each do |performer|
+      performer.told_jokes.each do |told_joke|
+        open_mic_jokes << told_joke
+      end
+    end
+    open_mic_jokes != open_mic_jokes.uniq
```

Comment:
this last line of code is a great use case, and very similar to what we'd want in the program